python/tvm/relay/op/transform.py

```diff
@@ -110,7 +110,17 @@ def expand_dims(data, axis, num_newaxis=1):
     result : relay.Expr
         The reshaped result.
     """
-    return _make.expand_dims(data, axis, num_newaxis)
+    if isinstance(axis, int):
+        return _make.expand_dims(data, axis, num_newaxis)
+    if isinstance(axis, Expr):
+        # TODO (AndrewZhaoLuo): investigate performance issues with consecutive
+        # dynamic expand_dims on non-llvm targets.
+        for _ in range(num_newaxis):
+            # Dynamic rank is not well supported so we can only increase rank
+            # by a static amount (e.g. 1) so we have to do this
+            data = _dyn_make.expand_dims(data, axis)
+        return data
+    raise ValueError(f"Unknown type for axis: {type(axis)}")
```

Comment:
Why can't we do it all at once if we know that num_newaxis is static?

Reply:
We can, done.